experimental/InjectiveResolutions/src/InjectiveResolutions.jl

```diff
@@ -0,0 +1,573 @@
+## Functions visible on the outside
+export mod_quotient
+export irreducible_res
+export get_all_ass_primes
+export quotient_by_ideal
+
+## Data 
+export FaceQ 
+
+#########################
+# some composite types
+#########################
+
+struct FaceQ # face of semigroup  
+    prime::Ideal 
+    poly::Polyhedron  #face of Q corresponding to prime
+    A::Matrix{Int64}  
+    b::Vector{Int64} 
+end
+
+struct IrrHull # irreducible hull
+    faces::Vector{FaceQ}
+    vectors::Vector{Tuple{SubquoModuleElem,Vector{Int}}}
+    lambda::AbstractAlgebra.Generic.MatSpaceElem
+end
+
+struct IndecInj # indecomposable injective
+    prime::Ideal
+    vector::Vector{Int}
+end
+
+struct IrrSum # irreducible sum
+    mod::SubquoModule
+    components::Vector{IndecInj}
+end
+struct IrrRes # irreducible resolution (including all computed data and the cochain complex)
+    irrSums::Vector{IrrSum}
+    cochainMaps::Vector{SubQuoHom}
+    surjections::Vector{SubQuoHom}
+    inclusions::Vector{SubQuoHom}
+    cokernels::Vector{SubquoModule}
+    cochainComplex::ComplexOfMorphisms # if sequence not exact return trivial cochain_complex (M0 -> M0)
+end
+
+##computes system of inequalities for polyhedron {r*a : r >= 0}, a \in ZZ^2
+##INPUT:    vector a in ZZ^2
+##OUTPUT:   matrix A, vector b such that Ax <= b for all x on hyperplane through a
+function compute_inequalities(a::Vector{Int})
+    num_variables = length(a)
+    num_inequalities = 2 * num_variables
+
+    A = zeros(Int, num_inequalities, num_variables)
+    b = zeros(Int, num_inequalities)
+    if !is_zero(a)
+        A[1,1] = -1
+        A[1,2] = 0
+        A[2,1] = 0
+        A[2,2] = -1
+        A[3,1] = a[2]
+        A[3,2] = -a[1]
+        A[4,1] = -a[2]
+        A[4,2] = a[1]
+    else
+        A[1,1] = -1
+        A[1,2] = 0
+        A[2,1] = 0
+        A[2,2] = -1
+        A[3,1] = 1
+        A[3,2] = 0
+        A[4,1] = 0
+        A[4,2] = 1
+    end
+    return A, b
+end
+
+@doc raw"
+    get_face_to_prime(P_F::Ideal)
+
+    Given a homogeneous prime ideal, return the corresponding face F
+
+    INPUT:    prime ideal p_F
+    OUTPUT:   list [face F, matrix A, vector b] such that F is defined by Ax <= b
+"
+function get_face_to_prime(P_F::Ideal)
+    R_Q = base_ring(P_F)
+    #get generators of k[F] = k[Q]/P_F
+    Q_F,_ = quo(R_Q,P_F)
+    G_F = gens(Q_F)
+    L_F = (P_F,[])
```

Comment:
Hmm, I am confused. You never access `L_F[1]`. Could you perhaps do this
```suggestion
    L_F = []
```
and then below replace `L_F[2]` by just `L_F`?

In addition `[]` is a `Vector{Any}` which isn't great for type stability.